R/utility_spliceHelper_functions.R

```diff
@@ -20,220 +20,336 @@
+                                                       strand)
+  annotatedTable$internalFirstExon.query <- annotateInternalStartEnd(queryStartRng,
+                                                                    subjectFullRng, annotatedTable$alternativeFirstExon)
+  annotatedTable$internalFirstExon.subject <- annotateInternalStartEnd(subjectStartRng,
+                                                                      queryFullRng, annotatedTable$alternativeFirstExon)
+  # TES/last exon
+  annotatedTable$alternativeLastExon <- alternativeStartEndExon(queryEndRng, 
+                                                               subjectEndRng)
+  annotatedTable$alternativeTES <- calculateTESdistance(queryEndRng, 
+                                                       subjectEndRng, 
+                                                       annotatedTable$alternativeLastExon,
+                                                       strand)
+  annotatedTable$internalLastExon.query <- annotateInternalStartEnd(queryEndRng,
+                                                                   subjectFullRng, annotatedTable$alternativeLastExon)
+  annotatedTable$internalLastExon.subject <- annotateInternalStartEnd(subjectEndRng,
+                                                                     queryFullRng, annotatedTable$alternativeLastExon)
+  #intron retention subject and query
+  annotatedTable$intronRetention.subject <- annotateIntronRetent(querySpliceRng, subjectFullRng)
+  annotatedTable$intronRetention.query <- annotateIntronRetent(subjectSpliceRng ,queryFullRng)
+  #exon skipping query and subject
+  annotatedTable$exonSkipping.query <- annotateExonSkip(querySpliceRng, subjectFullRng,
+                                                        subjectStartRng, subjectEndRng)
+  annotatedTable$exonSkipping.subject <- annotateExonSkip(subjectSpliceRng ,queryFullRng,
+                                                          queryStartRng, queryEndRng) 
+  #exon 5' and 3' splice site
+  exonSpliceTable <- annotateExonSplice(querySpliceRng, subjectFullRng,
+                                        subjectStartRng, subjectEndRng,
+                                        annotatedTable$strand)
+  annotatedTable <- cbind(annotatedTable, exonSpliceTable)
+  return(annotatedTable)
 }
 
+#' extract strand from GRangesList
+#' @description This function takes a GRangesList and
+#' returns a vector with the strand for each list entry.
+#' This function assumes that all elements for each list
+#' entry have the same strand.
+#' @usage getStrangeFromGrList(grl)
+#' @params grl a GRangesList
+#' @return an Rle object with strand information
+#' @examples
+#' query <- readRDS(system.file("extdata", 
+#'     "annotateSpliceOverlapByDist_testQuery.rds",
+#'     package = "bambu"))
+#' strand <- as.character(getStrandFromGrList(query))
+#' @noRd
+getStrandFromGrList <- function(grl) { 
+  return(unlist(strand(grl), use.names = F)[cumsum(elementNROWS(grl))]) 
```

Comment:
hi @yuukiiwa, you can use this line to get the strand 
`unique(unlist(strand(grl))`

Reply:
> Hi @yuukiiwa, this looks good to me! I had a few comments regarding the sub-functions, I generally feel that some of the sub-functions are quite similar, so instead of having one function for each case, have just one function with additional variable to indicate purpose might be a better way of forward, For details, please see my comments! Thanks!

Thanks for the review @cying111! The getStrandFromGrList was an idea from me, as this can be helpful for some other parts in the code where I used to store the strand so that I don't have to recalculate it. This calculation is fast, which makes it more efficient than storing the strand, so this might help make the code much simpler

Reply:
Hi @cying111, I do agree that the getStrandFromGrList function can be expressed as a command within the main compareTranscripts() function. The reason why we made this function was to make the main compareTranscripts() function cleaner to make it easier to be interpreted. If you think it is unnecessary I can replace the current function with the command.